### DISCUSSION TITLE
Delete alternative sections

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#502] [Q&A] [answered]
I noticed an inconvenience. If we create a route from point A to B, then according to the settings, Cruiser offers us alternatives. This is great. If we decide on one and delete the others, then when extending the route from B to C, alternatives are no longer proposed. As if the alternate individual sections were connected to each other. 

We also observe this when calculating a route with many intermediate points - deleting an alternative from one section results in deleting some of the remaining sections.

Is it possible to have a greater separation of alternatives between sections? This would make the route easier to read when creating complex ones where sections intersect with each other.

**Comment by devemux86** [comment]
> If we decide on one and delete the others, then when extending the route from B to C, alternatives are no longer proposed.

@Ciutnik thanks for the report, I will check it. :+1: 

> Is it possible to have a greater separation of alternatives between sections?

This would probably complicate enormously the implementation. :slightly_smiling_face: 
Note that routes are provided by routers and stored in all formats as whole routes: 2, 3, 4.

e.g. we calculate a route with many sections and get 3 alternative routes.
Delete 1 section, we still have 3 routes and would need to merge the rest 2 sections randomly with the 3 routes.

> This would make the route easier to read when creating complex ones where sections intersect with each other.

Cruiser already offers the best alternatives implementation:
freely tap to activate and combine any alternative sections into 1 master route.

Then we can drop the unwanted alternatives and use this best master route.

Doesn't this cover our needs?

**Reply by Ciutnik** [reply]
> I will check it.

Thanks! 

> Doesn't this cover our needs?

Generally yes! But with a complicated route, e.g. back and forth, we get a tangle of alternatives 😉 Sometimes it's hard to tell which one belongs to which section. In such a case, deleting one clearly unnecessary alternative may lead to deleting one needed in another section. I was thinking about a some marker assigning alternatives to individual sections, but I don't think we want another graphic element on the map/route?

As I mentioned, the best way is to choose the best alternative and remove unnecessary ones when adding new viapoints - section by section... But here we go back to the opening post on this topic, i.e. the lack of proposals for alternatives in the newly created sections 🙂

**Reply by devemux86** [reply]
> the lack of proposals for alternatives in the newly created sections

I can reproduce it with the extend function and will check it.

> the best way is to choose the best alternative and remove unnecessary ones when adding new viapoints - section by section

I can see what is possible.

Current workflow is select the wanted alternative section to connect it with the main route and continue the planning.
At the end, delete the not needed alternative routes and use the main route.

**Reply by Ciutnik** [reply]
I agree, it works well, but in **specific** situations (and I have a problem with this one, very unpleasant because it was related to my friend's funeral) we get nystagmus 😉 This is where planning and removing section by section would be ideal 👍

**Comment by devemux86** [comment]
@Ciutnik 

1. This will work properly in the next update. Thanks for the report!

**Reply by Ciutnik** [reply]
Thank you very much! You saved my eyes! I no longer suffer from nystagmus 👀👍

**Comment by devemux86** [comment]
2. This is also implemented, now it is tested.

There will be a new menu "Delete section" on the alternative sections.

We are looking for 1 or 2 icons from [here](https://fonts.google.com/icons?icon.set=Material+Icons&icon.style=Filled) for the menus:
- Delete section
- Delete route (now has the [Delete Forever](https://fonts.google.com/icons?selected=Material+Icons:delete_forever:&icon.query=delete&icon.set=Material+Icons&icon.style=Filled) icon)

**Reply by devemux86** [reply]
If we keep the [Delete Forever](https://fonts.google.com/icons?selected=Material+Icons:delete_forever:&icon.query=delete&icon.set=Material+Icons&icon.style=Filled) icon for the "Delete route" menu
we could use the [Delete Outline](https://fonts.google.com/icons?selected=Material+Icons:delete_outline:&icon.query=delete&icon.set=Material+Icons&icon.style=Filled) for the "Delete section" menu.

Or select 2 new icons for both menus.

**Reply by Ciutnik** [reply]
Has the [Remove road](https://fonts.google.com/icons?selected=Material%20Icons%3Aremove_road%3A) icon been used somewhere before?

![SmartSelect_20240421_181423_Chrome](https://github.com/devemux86/cruiser/assets/122647782/5c3bfa07-29f0-46de-ac07-259347c8a23c)

**Reply by devemux86** [reply]
In this menu, no.
But shouldn't these 2 delete icons be a bit related?

(like the 2 recalculate section + route icons)

**Reply by Ciutnik** [reply]
> But shouldn't these 2 delete icons be a bit related?

Yes, indeed. I don't think there's much choice here, so all you have to do is use two different "waste bin" icons.

**Reply by Ciutnik** [reply]
If it's a vector graphic, maybe you can overlay two icons? The "upper" one would just have to have some outline to stand out nicely from the lower one. (Or possibly use the main icon and a second, smaller one somewhere in the corner?)

Combining icons would provide unlimited possibilities for creating new, more understandable graphics in the future.

**Reply by bios4** [reply]
My proposal would be the "Delete Outline" (optionally the "Delete" icon w/o the "x") or the "Delete Sweep" icon!
Somehow the latter could look nice I think.

Or simply the "Clear" icon, but if you want to have them looking related I'd go with one of the first three of these ^^...

**Comment by devemux86** [comment]
A new function to delete sections (in alternative routes) is now available in the latest:
- https://github.com/devemux86/cruiser/discussions/482

(available in route menu)

**Comment by devemux86** [comment, marked as answer]
Delete sections (in alternative routes) is available in [Cruiser 4.0.0](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
